app/src/main/java/com/example/caloriecounter/feature/restaurant/ui/meals/MealsViewModel.kt

```diff
@@ -0,0 +1,59 @@
+package com.example.caloriecounter.feature.restaurant.ui.meals
+
+import androidx.lifecycle.viewModelScope
+import androidx.navigation.NavBackStackEntry
+import com.example.caloriecounter.base.BaseViewModel
+import com.example.caloriecounter.feature.restaurant.ui.RestaurantScreen
+import com.example.domain.handle
+import com.example.domain.usecases.retrofit.GetMealListUseCase
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+class MealsViewModel @Inject constructor(
+    private val getMealListUseCase: GetMealListUseCase,
+    private val backStackEntry: NavBackStackEntry,
+) : BaseViewModel<MealsContract.Event, MealsContract.State, MealsContract.Effect>() {
+
+    private var category = ""
```

Comment:
If it's used in init you can probably directly put this code 
```
category = backStackEntry.arguments
            ?.getString(RestaurantScreen.MealList().argument) ?: ""
```

Reply:
Ideally you can create a factory for your ViewModel so you can create a new MealsViewModel with Factory in it

Reply:
@yaroslav-android correct me if I am wrong

Reply:
Why do we need that? We can use [SavedStateHandle](https://developer.android.com/topic/libraries/architecture/viewmodel-savedstate)